internal/controllers/addon/handle_installed_condition.go

```diff
@@ -0,0 +1,65 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+func (r *olmReconciler) handleInstalledCondition(ctx context.Context,
+	addon *addonsv1alpha1.Addon, addonCSVRef *operatorsv1.RichReference) (requeueResult, error) {
+	// Handle missing CSV, addon might have been uninstalled?.
+	if addonCSVRef == nil {
+		return r.handleMissingCSV(ctx, addon)
+	}
+	csvPhase := getCSVPhase(addonCSVRef)
+	// If csv is in the succeeded phase and the last observed CSV is empty, we know that
+	// this CSV is coming up for the first time.
+	if csvPhase == operatorsv1alpha1.CSVPhaseSucceeded && addon.Status.LastObservedAvailableCSV == "" {
+		reportInstalledCondition(addon)
+	}
+	return resultNil, nil
```

Comment:
> What about the case if csvPhase == operatorsv1alpha.CSVPhaseSucceeded irrespective of addon.Status.LastObservedAvailableCSV, shouldn't we report something as well, instead of leaving as it is?

The Available condition precisely captures that scenario.

Reply:
Installed condition is set to true when the addon is first "installed" after that it is left untouched.

Reply:
hmm, I don't like the "untouched" aspect of this. There should be a way to reconcile that condition back whenever that's required. :/ 

I'd say something like the following should happen 

```
	if csvPhase == operatorsv1alpha1.CSVPhaseSucceeded {
		reportInstalledCondition(addon)
	}
```
This clearly indicates that if the CSV was ever succeeded, it should report "Installed" condition as true, which "actually" is the case. In our ecosystem, CSV readiness is treated as "installed".
Beyond that, the combination of "installed = true" and "available: true/false", whether the addon was successfully installed or not, OR the addon failed after a successful installation

Reply:
Yeah that makes more sense, the untouched aspect of it doesn't allow us to reconcile that status back. 
Good catch 👍